aws-java-sdk-sns/src/main/java/com/amazonaws/services/sns/util/Topics.java

```diff
@@ -186,8 +186,9 @@ public static String subscribeQueue(AmazonSNS sns, AmazonSQS sqs, String snsTopi
         Map<String, String> sqsAttrs = sqs.getQueueAttributes(sqsQueueUrl, sqsAttrNames).getAttributes();
         String sqsQueueArn = sqsAttrs.get(QueueAttributeName.QueueArn.toString());
 
-        Policy policy = extendPolicy ? Policy.fromJson(
-                sqsAttrs.get(QueueAttributeName.Policy.toString())) : new Policy();
+        String policyJson = sqsAttrs.get(QueueAttributeName.Policy.toString());
+        Policy policy = extendPolicy || policyJson == null || policyJson.length() == 0
```

Comment:
I have several environments and I guess I got confused. I will correct it. Thanks a lot.